packages/database/src/cachers/EntityHierarchyCacher.js

```diff
@@ -51,28 +160,12 @@ export class EntityHierarchyCacher {
       useEntityRelationLinks,
     );
 
-    const childrenAlreadyCached = await this.checkChildrenAlreadyCached(
-      hierarchyId,
-      parentIds,
-      childCount,
-    );
-    if (!childrenAlreadyCached) {
-      await this.cacheGeneration(hierarchyId, childIdToAncestorIds);
-    }
+    await this.cacheGeneration(hierarchyId, childIdToAncestorIds);
```

Comment:
After writing more tests, I realised that the previous coarse grained approach of "if the generation isn't perfectly cached, recache the whole generation" would cause double ups. Now it is _only_ recaching the bits that aren't already cached (see the `cacheGeneration` function)